File: testdata/bench/go/BenchmarkTest00004.go
package bench

import (
	"database/sql"
	"net/http"
)

// SQL injection via channel propagation
func Handler00004(w http.ResponseWriter, r *http.Request) {
	param := r.FormValue("search")
	ch := make(chan string, 1)
	ch <- param
	tainted := <-ch
	db, _ := sql.Open("sqlite3", ":memory:")
	db.Query("SELECT * FROM items WHERE name = '" + tainted + "'")
}

File: testdata/bench/kotlin/BenchmarkTest00004.kt
package bench

import javax.servlet.http.HttpServletRequest
import java.sql.DriverManager

// Vulnerable: SQL injection via servlet request parameter
fun handler00004(request: HttpServletRequest) {
    val id = request.getParameter("id")
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    conn.createStatement().executeQuery("SELECT * FROM orders WHERE id = $id")
}
